internal/network/data_sender.go

```diff
@@ -291,19 +307,6 @@ func (ds *dataSender) sendPayload(topic string, payload []byte, compress bool) e
 		return fmt.Errorf("payload did not have expected subject cloud event property")
 	}
 
-	// Connect to the MQTT broker
```

Comment:
I don't know the intention of connecting to the broker each time we send a message and disconnecting after the message is sent(maybe because reconnect was disabled?). It seems unnecessary overhead for me.

If we set opts.SetConnectRetry(true), the client will automatically try to reconnect the lost connection.